packages/flutter_test/lib/src/matchers.dart

```diff
@@ -352,6 +385,185 @@ class _HasOneLineDescription extends Matcher {
   Description describe(Description description) => description.add('one line description');
 }
 
+class _NormalizeHashCodesEquals extends Matcher {
+  _NormalizeHashCodesEquals(String v) : _value = _normalize(v);
+
+  final String _value;
+
+  static final Object _mismatchedValueKey = new Object();
+
+  static String _normalize(String s) {
+    return s.replaceAll(new RegExp(r'#[0-9a-f]{5}'), '#000');
+  }
+
+  @override
+  bool matches(dynamic object, Map<dynamic, dynamic> matchState) {
+    final String description = _normalize(object);
+    if (_value != description) {
+      matchState[_mismatchedValueKey] = description;
+      return false;
+    }
+    return true;
+  }
+
+  @override
+  Description describe(Description description) {
+    return description.add('multi line description equals $_value');
+  }
+
+  @override
+  Description describeMismatch(
+    dynamic item,
+    Description mismatchDescription,
+    Map<dynamic, dynamic> matchState,
+    bool verbose
+  ) {
+    if (matchState.containsKey(_mismatchedValueKey)) {
+      final String actualValue = matchState[_mismatchedValueKey];
+      // Leading whitespace is added so that lines in the multi-line
+      // description returned by addDescriptionOf are all indented equally
+      // which makes the output easier to read for this case.
+      return mismatchDescription
+          .add('expected normalized value\n  ')
+          .addDescriptionOf(_value)
+          .add('\nbut got\n  ')
+          .addDescriptionOf(actualValue);
+    }
+    return mismatchDescription;
+  }
+}
+
+/// Returns `true` if [c] represents a whitespace code unit.
```

Comment:
I would put all these inside the class as static methods so that it's all wrapped up together